### PR TITLE
[DUOS-3062][DUOS-3064][risk=no] Refactor Data Use validation and translation for V4

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-3064
Should also address https://broadworkbench.atlassian.net/browse/DUOS-3062 after a dataset re-index.

## Summary
Significant refactor of all data use related code
* Remove the oldest v2 match logic - now returns an `HTTP 410 GONE` response
* Deprecate the next oldest v3 match logic. It is still functional, just marked as deprecated.
* Update all translation logic to match latest v4 logic
* Refactor schema validation for v4 schema
* **_Dev Note_**: I chose not to run v4 schema validation for the translation endpoint. We do have a validation endpoint, so the schema can be validated that way, but for translation purposes, I think it is adequate to return a translation based on what is provided as best we can with the available elements of the schema.

---
There is some new code here that didn't exist in the previous translations:

* **Publication Moratorium** was not captured previously. It is now shown as `"Publishing moratorium until '%s' is in effect. [MOR]"`
* **Secondary Other** was not captured previously. It is now shown as `"Secondary other restrictions: %s."`
* **Population** was not captured previously, it is now shown as `"Future use for study variation in the general population (e.g. calling variants and/or studying their distribution). [POP]"`

The following terms can be used in algorithmic decisions and result in an **[ABSTAIN](https://github.com/DataBiosphere/consent-ontology/blob/develop/docs/Algorithms.md)** decision. They are not captured on dataset/study ingest and are therefore not used in translations:
* `illegalBehavior`
* `sexualDiseases`
* `stigmatizeDiseases`
* `vulnerablePopulations`
* `psychologicalTraits`
* `notHealth`

#### `{ "generalUse": true }`
```
{
  "primary": [
    {
      "code": "GRU",
      "description": "Data is available for general research use."
    }
  ],
  "secondary": []
}
```
#### `{ "diseaseRestrictions": ["http://purl.obolibrary.org/obo/DOID_162"] }`
```
{
  "primary": [
    {
      "code": "DS",
      "description": "Data use is limited for studying: cancer"
    }
  ],
  "secondary": []
}
```
#### `{ "hmbResearch": true }`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    }
  ],
  "secondary": []
}
```
#### `{ "hmbResearch": true, "populationOriginsAncestry": true }`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    },
    {
      "code": "NPOA",
      "description": "Future use for population origins or ancestry research is prohibited."
    }
  ],
  "secondary": []
}
```
#### `{ "hmbResearch": true, "other": "other" }`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    },
    {
      "code": "OTHER",
      "description": "Other restrictions: other."
    }
  ],
  "secondary": []
}
```
#### `{ "hmbResearch": true, "methodsResearch": true }`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    }
  ],
  "secondary": [
    {
      "code": "NMDS",
      "description": "Data use for methods development research ONLY within the bounds of other data use limitations."
    }
  ]
}
```
#### `{ "hmbResearch": true, "nonProfitUse": true }`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    }
  ],
  "secondary": [
    {
      "code": "NCU",
      "description": "Commercial use prohibited."
    }
  ]
}
```
#### `{ "hmbResearch": true, "secondaryOther": "other 2"}`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    }
  ],
  "secondary": [
    {
      "code": "OTHER",
      "description": "Secondary other restrictions: other 2."
    }
  ]
}
```
#### `{ "hmbResearch": true, "ethicsApprovalRequired": true }`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    }
  ],
  "secondary": [
    {
      "code": "IRB",
      "description": "Local ethics committee approval is required."
    }
  ]
}
```
#### `{ "hmbResearch": true, "collaboratorRequired": true }`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    }
  ],
  "secondary": [
    {
      "code": "COL",
      "description": "Collaboration with the primary study investigators required."
    }
  ]
}
```
#### `{ "hmbResearch": true, "geographicalRestrictions": "geography" }`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    }
  ],
  "secondary": [
    {
      "code": "GS",
      "description": "Geographical restrictions: geography."
    }
  ]
}
```
#### `{ "hmbResearch": true, "geneticStudiesOnly": true }`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    }
  ],
  "secondary": [
    {
      "code": "GSO",
      "description": "Future use is limited to genetic studies only"
    }
  ]
}
```
#### `{ "hmbResearch": true, "publicationResults": true }`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    }
  ],
  "secondary": [
    {
      "code": "PUB",
      "description": "Publishing results of studies using the data available to the larger scientific community is required"
    }
  ]
}
```
#### `{ "hmbResearch": true, "publicationMoratorium":"01/01/2000" }`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    }
  ],
  "secondary": [
    {
      "code": "MOR",
      "description": "Publishing moratorium until '01/01/2000' is in effect."
    }
  ]
}
```
#### `{ "hmbResearch": true, "controls": true }`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    }
  ],
  "secondary": [
    {
      "code": "NCTRL",
      "description": "Future use as a control set for diseases other than those specified is prohibited."
    }
  ]
}
```
#### `{ "hmbResearch": true, "gender": "Male" }`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    }
  ],
  "secondary": [
    {
      "code": "POP-M",
      "description": "Data use is limited to research on males."
    }
  ]
}
```
#### `{ "hmbResearch": true, "gender": "Male" }`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    }
  ],
  "secondary": [
    {
      "code": "POP-F",
      "description": "Data use is limited to research on females."
    }
  ]
}
```
#### `{ "hmbResearch": true, "pediatric": true }`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    }
  ],
  "secondary": [
    {
      "code": "POP-PD",
      "description": "Data use is limited to pediatric research."
    }
  ]
}
```
#### `{ "hmbResearch": true, "population": true }`
```
{
  "primary": [
    {
      "code": "HMB",
      "description": "Data is limited for health/medical/biomedical research."
    }
  ],
  "secondary": [
    {
      "code": "POP",
      "description": "Future use for study variation in the general population (e.g. calling variants and/or studying their distribution)."
    }
  ]
}
```

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/dsde/consent/ontology/datause/services/TextTranslationServiceImpl.java [hunk 1]
```diff
@@ -29,49 +28,27 @@
...
+  private static final String GRU = "Data is available for general research use. [GRU]";
+  private static final String DS = "Data use is limited for studying: %s [DS]";
+  private static final String HMB = "Data is limited for health/medical/biomedical research. [HMB]";
+  private static final String POA = "Future use for population origins or ancestry research is prohibited. [POA]";
+  private static final String NMDS = "Data use for methods development research ONLY within the bounds of other data use limitations. [NMDS]";
+  private static final String NCU = "Commercial use prohibited. [NCU]";
+  private static final String OTHER = "Other restrictions: %s.";
+  private static final String SECONDARY_OTHER = "Secondary other restrictions: %s.";
   private static final String ETHICS_APPROVAL = "Local ethics committee approval is required.";
+  private static final String COLLABORATION_REQUIRED = "Collaboration with the primary study investigators required.";
```
Comment (fboulnois) [inline src/main/java/org/broadinstitute/dsde/consent/ontology/datause/services/TextTranslationServiceImpl.java]:
```suggestion
  private static final String COLLABORATION_REQUIRED = "Collaboration with the primary study investigators required. [COL]";
```

**Review by fboulnois — APPROVED** [review]
This cleanup looks great! Thank you. One small nit:

**Review by aarohinadkarni — COMMENTED** [review]
When running locally, the payload 
`{ "diseaseRestrictions": ["http://purl.obolibrary.org/obo/DOID_162"] }` is returning 
`can't parse JSON.  Raw result: Error while translating: HTTP 500 Internal Server Error` as opposed to the response specified in the PR description.

**Comment by rushtong** [comment]
> When running locally, the payload `{ "diseaseRestrictions": ["http://purl.obolibrary.org/obo/DOID_162"] }` is returning `can't parse JSON. Raw result: Error while translating: HTTP 500 Internal Server Error` as opposed to the response specified in the PR description.

@aarohinadkarni - With the input you have quoted here, I get back the following response:
```
{
  "primary": [
    {
      "code": "DS",
      "description": "Data use is limited for studying: cancer"
    }
  ],
  "secondary": []
}
```
One thing I do see is that we get this when there is no connection to Elastic Search. When locally testing, you will need to be on the VPN. I'd like to tackle this separately since it is an existing incorrect error response.

**Review by aarohinadkarni — APPROVED** [review]
looks great! 👍

**Pull request merged**